### DISCUSSION TITLE
Is this project dead?

### DISCUSSION THREAD

**Discussion by mjablecnik** [isar/isar#118] [Show and tell] [answered]
Hello @leisim,
from first see of your post in Reddit here: https://www.reddit.com/r/FlutterDev/comments/meedy7/isar_database_fast_acid_compliant_nosql_database/

I look forward to first stable version of isar database but now I see that for last 3 months is not any new commit here in isar repository..
What does it mean? I thought that isar will be early ready from september but now I don't see any activity here.

Is this project dead or what are plans for the future of this project?

Thank you for your response.

**Comment by MarcelGarus** [comment]
There have been commits yesterday, just not in this main repository: https://github.com/isar/isar-core/commit/8e01b22a5020b3334a180c6edb63b635d3fce57c

**Reply by simc** [reply]
I'm working on a huge commit 😅

**Reply by obalak** [reply]
Hi Simon, 
As I'm more used to NoSQL DBs (Couchbase and mongodb) on backend development, I'm struggling to choose the right database for our flutter desktop app and I have seen recommendations to use Isar from Hive developer and I would like to avoid using Sqflite or in general SQLite. My question, is the project being actively developed or is it in standby. I fully understand that developing advanced features and tests can take longtime ... 
Any news/direction?
Appreciate your answer.

**Comment by simc** [comment, marked as answer]
@mjablecnik I haven't committed any code recently but the project is far from dead and I'm working on a massive feature update. Also still fighting with unit tests haha.

**Reply by mjablecnik** [reply]
@leisim Ok thank you very much.. I was very concerned what happened because it is more then 4 months and I didn't see any activity here or in any other branch.. :)
So I hope that we will can see stable release early and thank you very much for your response. ;)

**Reply by yringler** [reply]
@leisim you're too much!
>  still fighting with unit tests

_So I guess I'll just go and write [my own testing framework](https://github.com/leisim/honey).
That's what people do, right?_

You are legend; you bring to mind Linus Torvalds's "I remember when men where men and wrote their own device drivers"

For now I'm using moor (now drift), but keeping an eye on Isar.

**Reply by simc** [reply]
@yringler Hahaha you are right. Flutter is really missing test frameworks and it had very high priority for me since I needed it urgently.

Honey is an end to end testing framework and allows you to write tests like this:

```
tap the "Login" button

verify that the text field below the "logo" image is empty

enter "username" into it
```

But I will need Isar very soon for my work so I'll continue working on it after Honey is finished.

**Reply by bambinoua** [reply]
@simc your answer was 3 years ago and last commit 8 months ago. Should we still wait on project activity?

**Comment by stevenkissack** [comment]
@leisim that's great news, we're personally starting a rather large flutter app next week and are currently picking between Isar, Moor & sqflite. 

Are we likely to see any breaking changes that require dumping collections/tables if we were to go with Isar? I think it'd be my head on the chopping block if we broke every app in production, although I appreciate it's currently alpha so we'd have to keep an eye on releases etc.

Any updates or comments/warnings for us and the community would be amazing, personally I think Isar is the clear choice when needing Stream UI updates and a NoSQL backing and generated types - nobody likes complex migrations on user devices..

I didn't see a mention of shipping an existing Isar database with an app release, or allowing a binary file download of a database to a phone? That'd be an amazing feature if we can create that somehow on the server without needing a Dart backend (similar to how we would do it with sqlite)

Edit: Probably easy to work around that import db limitation by just holding large json blobs :)

Appreciate all your work 1000x

**Reply by simc** [reply]
Yes, unfortunately, the move to a stable release will come with breaking changes. Sorry 😕

**Comment by mjablecnik** [comment]
@leisim Hello Simon :) after half of year I have to ask you how you continue with Isar? Did you already solved your problems with tests?
When you expect that you can release your first stable or beta release of Isar?
I am asking because I would like to know if I can use it for my next new project or not. :)
Thank you for your response.

**Reply by simc** [reply]
Hey @mjablecnik, my plan is to release a stable version in 2021. It's looking good currently but I don't want to make any promises 😅

**Reply by mjablecnik** [reply]
@leisim That's unbelievable 😄 You really released it in 2021 year yet 😆😅
![image](https://user-images.githubusercontent.com/2686906/147842128-def0a851-91a4-4405-928a-2b88406a1b92.png)

Great work man 👍

**Reply by simc** [reply]
@mjablecnik It was close but it worked out haha

**Reply by CodeDoctorDE** [reply]
it depends on the time zone you're living

**Comment by Silfalion** [comment]
Hey leisim! Thank you so much for all the work you've done. I'm in the process of choosing a DB package for my open source project and was wondering if you plan to update isar often at this stage of the development.

**Reply by simc** [reply]
@Silfalion I plan to actively maintain it since I need it myself soon. I put a lot of work into it.

**Reply by Silfalion** [reply]
Awesome. And it shows you can't make something so simple and elegant without having a lot of work behind it. Then I think I'll add it to the next test and if everything goes well, will put it as the main DB package.

**Comment by Davut-Atajanov** [comment]
Hello, people. Is the project actually dead right now?

I have seen some people transitioning into the community version of ISAR. Do you think I should continue from there? According to GitHub stars ISAR is more popular than sqflite, it would be a pity if it is discontinued

**Reply by wish4129** [reply]
I agreed, it's actually disappointed that I really wish this repo to be huge success when hits v4. I don't see any movement near to half a year ago.

**Reply by Davut-Atajanov** [reply]
yes mate. I don't know if anybody else is looking for an alternative, but this might be helpful to some.

I decided to continue with ObjectBox. They are backed by company, 1k+ stars on github, and at least 10x faster than sqflite. Hive and Isar are not supported anymore. Drift is just built on top of sqflite so it uses SQL and way (WAY) slower than ObjectBox.

**Reply by felipebonezi** [reply]
I'm considering to remove Isar because the lack of support as well.
